src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_IsByRefLike.cs

```diff
@@ -636,12 +636,203 @@ public class Test
                 );
         }
 
-        private static void AssertReferencedIsByRefLikeAttribute(Accessibility accessibility, ImmutableArray<CSharpAttributeData> attributes, string assemblyName)
+        [Fact]
+        public void IsByRefLikeObsolete()
+        {
+            var text = @"
+namespace System.Runtime.CompilerServices
+{
+    public class IsByRefLikeAttribute : System.Attribute { }
+}
+
+class Test
+{
+    [System.Obsolete(""hello"", true)]
+    public ref struct S1 {}
+}
+";
+
+            CompileAndVerify(text, verify: false, symbolValidator: module =>
+            {
+                var type = module.ContainingAssembly.GetTypeByMetadataName("Test").GetTypeMember("S1");
+                Assert.True(type.IsByRefLikeType);
+
+                var accessibility = Accessibility.Public;
+                var attributes = type.GetAttributes();
```

Comment:
`Assert.Equal(2, attributes.Length)` to verify we're not adding emitting an additional `[Obsolete]`.